### DISCUSSION TITLE
MDEditor > codeMirror

### DISCUSSION THREAD

**Discussion by beckyi** [imzbf/md-editor-rt#176] [Q&A]
Hello Can I access codeMirrorUt instance? (for using MDEditor > codeMirror API)

I'd like to get command line number, so I need a codeMirrorUt instance. 
(view.lines, view.lineBlockAt(line().from).top, ...)

When scrolling down, it is difficult to see the entire line information in the dom.

```
//cc. scroll-auto.ts
const scrollAuto = (pEle: HTMLElement, cEle: HTMLElement, codeMirrorUt: CodeMirrorUt) => {
  const { view } = codeMirrorUt;

  const smoothScroll = createSmoothScroll();

  const getTopByLine = (line: number) => {
    return view.lineBlockAt(view.state.doc.line(line + 1).from).top;
  };

  const getBottomByLine = (line: number) => {
    return view.lineBlockAt(view.state.doc.line(line + 1).from).bottom;
  };

```

OR, add attributes remain line number in '.cm-line' element (command line)

```
EditorView.contentAttributes.of({"data-row-number": index })
```

cc. using version "md-editor-rt": "4.13.3"

**Comment by imzbf** [comment]
Will the following example meet your requirements?

```js
editorRef.current?.domEventHandlers({
  scroll(event, view) {
    console.log('scroll', view.lineBlockAtHeight(view.scrollDOM.scrollTop));
  }
});
```

**Comment by beckyi** [comment]
Thanks But your suggestion works to get the line value only when the editor scroll event occurs.
I want to get values ​​from different events. 
Could you please take a look at the proposal([PR](https://github.com/imzbf/md-editor-rt/pulls))? 🙏🏻

**Comment by imzbf** [comment]
Use `getEditorView` in `^4.20.0`

**Reply by beckyi** [reply]
@imzbf  thanks a  lot 🙇🏻‍♂️

But I meet new error... could you confirm this error?

cc. https://github.com/imzbf/md-editor-rt/issues/179
